Context menu styling + fixes

xoascf opened the pull request:
Taken mainly from an additional patch from July 2022; context menu styling: https://github.com/xoascf/RetroBar/commit/6c4e6cec4d6a859273cde99446ee8bd633eb5489, and others from 2023.

xoascf committed:
Context menu styling

dremin commented:
@xoascf Are any of these PRs in draft status ready to be looked at? I had assumed that you were still working on them but I might be wrong.

dremin commented:
Ran into a couple issues testing this:
- The styling of context menus doesn't seem to be quite right
![Screenshot 2023-02-02 at 11 02 04 PM](https://user-images.githubusercontent.com/7373633/216516604-272dbdba-9f74-42a6-a574-fef194f69484.png)

- Tile horizontally seems broken?
![Screenshot 2023-02-02 at 11 01 55 PM](https://user-images.githubusercontent.com/7373633/216516577-b11edb83-1b96-4fe6-b66f-e9ced957218e.png)

som3rando commented:
Also undoing sometimes causes retrobar to crash
[crash log.log](https://github.com/dremin/RetroBar/files/10582581/crash.log.log)

xoascf committed:
ButtonFlashingForeground → HighlightForeground

som3rando commented:
I found a few issues running this fork
On windows 11 insider (build 25314.1010) the undo feature is broken instead of undoing it opens a empty explorer window that is not removable until explorer shuts down(on windows 10 its fine).
[log1.log](https://github.com/dremin/RetroBar/files/11327469/log1.log)
https://youtu.be/x9luV5QVoYU
On windows 7 and 8 the undo feature is also broken but still semi funktional it does it job but it returns 2 of 3 windows back to the original state.
[log2.log](https://github.com/dremin/RetroBar/files/11327477/log2.log)
https://youtu.be/-6NjJvGQQIs

xoascf committed:
Context menu sound

xoascf committed:
Trying to decouple `TaskButtonContextMenu` to make it fully customizable (1)

xoascf committed:
Keep ContextMenu open after pressing Alt

xoascf committed:
Handle Alt+F4 key press to close the task

xoascf committed:
Add `InputGestureText` for Close menu item to indicate Alt+F4 shortcut

xoascf committed:
Handle system close command to open Shut Down Windows dialog box

Just like the stock taskbar does.

TheDolphinSlayer commented:
So glad this is still being worked on!

xoascf committed:
Test

home-gihub commented:
Time flies

xoascf committed:
Refactor text rendering mode handling

Replace per-element XAML bindings using BoolToTextRenderingModeConverter with centralized event handlers in App.xaml.cs. Windows now get TextRenderingMode applied via a class-level Loaded handler, and context menus via the existing MenuOpened handler. The BoolToTextRenderingModeConverter class is removed as it's no longer needed.

xoascf committed:
Fix task menu handlers and restore End Task item

Detach existing click handlers before reattaching task button system menu actions to prevent duplicate invocations when the menu is rebuilt. Also add an End Task menu entry in the system menu XAML (just like we had before) and wire it up in code-behind, with visibility controlled by ShowEndTaskButton.
